core/lib/dal/migrations/20240522081114_create_data_availability_table.up.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE data_availability
+(
+    l1_batch_number BIGINT PRIMARY KEY REFERENCES l1_batches (number) ON DELETE CASCADE,
+
+    -- the BYTEA used for this 2 columns because it is the most generic type
+    -- the actual format of blob identifier and inclusion data is defined by the DA client implementation
+    blob_id         BYTEA     NOT NULL, -- blob here is an abstract term, unrelated to any DA implementation
+    inclusion_data  BYTEA,
```

Comment:
Dumb question: Is storing `inclusion_data` in Postgres required because the DA client isn't expected to retain it forever, or is it just an optimization, or are there other reasons? I.e., would an approach work in which the client is queried each time we need `inclusion_data` for a batch (supposedly for sending it to `commitBatches` if I understand the design doc correctly)?

Reply:
it's not necessarily required, you are right that we can always call the DA client for each batch, it's mostly done for the convenience of having the data in the DB for debugging purposes, and decoupling of the direct dependency of the `eth_sender` on `da_dispatcher` in the case when we pass the data directly without storing it in the DB

and it's relatively low cost, I don't expect the inclusion data to be large